### PR TITLE
[Feature] New Parser and Analyzer support admin show syntax

### PR DESCRIPTION
**Pull request by dreamay**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Problem Summary(Required) ：

- Support admin show syntax
- Replace cup with antlr4

### CODE REVIEW

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 13 / 17 (76.47%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminShowConfigStmtAnalyzer.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 63 / 82 (76.83%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|4|6|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|39|55|70.91%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 110 / 140 (78.57%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|72|97|74.23%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 118 / 140 (84.29%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|80|97|82.47%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 120 / 142 (84.51%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|82|99|82.83%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AdminStmtAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                return false;
+            }
+
+            BinaryPredicate binaryPredicate = (BinaryPredicate) where;
+            BinaryPredicate.Operator op = binaryPredicate.getOp();
+            if (op != BinaryPredicate.Operator.EQ && op != BinaryPredicate.Operator.NE) {
+                return false;
+            }
+
+            Expr leftChild = binaryPredicate.getChild(0);
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AdminStmtAnalyzer.java:177]: These if statement is not so intuitive, do we have better way to resolve it?

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -40,9 +40,13 @@ statement
     | updateStatement                                                                       #update
     | deleteStatement                                                                       #delete
 
-    // Admin Set Statement
+    // Admin Statement
     | ADMIN SET FRONTEND CONFIG '(' property ')'                                            #adminSetConfig
     | ADMIN SET REPLICA STATUS properties                                                   #adminSetReplicaStatus
+    | ADMIN SHOW FRONTEND CONFIG (LIKE pattern=string)?                                     #adminShowConfig
+    | ADMIN SHOW REPLICA DISTRIBUTION FROM qualifiedName (PARTITION identifierList)?        #adminShowReplicaDistribution
+    | ADMIN SHOW REPLICA STATUS FROM qualifiedName (PARTITION identifierList)?
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:48]: Shall we add a show help command to display all these commands?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AdminStmtAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.sql.ast.AstVisitor;
+
+import java.util.List;
+import java.util.Map;
+
+public class AdminStmtAnalyzer {
+    public static void analyze(AdminSetConfigStmt adminSetConfigStmt, ConnectContext session) {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AdminStmtAnalyzer.java:27]: Each statement should not implement an analyze statement, it should implement only one, the parameter can be StatementBase, and then the Visitor calls the corresponding visitXXX according to the specific type

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 2 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:48]: Help statements can be provided, but that shouldn't be the job of this PR. And, currently our help command framework has been under maintenance

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 103 / 125 (82.40%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|65|82|79.27%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java [hunk 4]
```diff
@@ -89,6 +92,34 @@ public Void visitAdminSetReplicaStatusStatement(AdminSetReplicaStatusStmt statem
...
+            // check auth
+            if (!GlobalStateMgr.getCurrentState().getAuth().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "ADMIN");
+            }
+            return null;
+        }
+
+        @Override
+        public Void visitAdminShowReplicaStatusStatement(AdminShowReplicaStatusStmt statement, ConnectContext session) {
+            // check auth
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java:116]:
```suggestion
```

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 5]
```diff
@@ -44,9 +44,13 @@ statement
     | updateStatement                                                                       #update
     | deleteStatement                                                                       #delete
 
-    // Admin Set Statement
+    // Admin Statement
     | ADMIN SET FRONTEND CONFIG '(' property ')'                                            #adminSetConfig
     | ADMIN SET REPLICA STATUS properties                                                   #adminSetReplicaStatus
+    | ADMIN SHOW FRONTEND CONFIG (LIKE pattern=string)?                                     #adminShowConfig
+    | ADMIN SHOW REPLICA DISTRIBUTION FROM qualifiedName (PARTITION identifierList)?        #adminShowReplicaDistribution
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:51]:
```suggestion
    | ADMIN SHOW REPLICA DISTRIBUTION FROM qualifiedName partitionNames?        #adminShowReplicaDistribution
```

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AdminStmtAnalyzer.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+        public Void visitAdminShowReplicaDistributionStatement(
+                AdminShowReplicaDistributionStmt adminShowReplicaDistributionStmt,
+                ConnectContext session) {
+            String dbName = adminShowReplicaDistributionStmt.getDbName();
+            String tblName = adminShowReplicaDistributionStmt.getTblName();
+            if (Strings.isNullOrEmpty(dbName)) {
+                if (Strings.isNullOrEmpty(session.getDatabase())) {
+                    ErrorReport.reportSemanticException(ErrorCode.ERR_NO_DB_ERROR);
+                } else {
+                    dbName = ClusterNamespace.getFullName(session.getClusterName(), session.getDatabase());
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AdminStmtAnalyzer.java:83]: setDb return to adminShowReplicaDistributionStmt

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 101 / 123 (82.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|65|82|79.27%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|17|17|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 107 / 131 (81.68%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|67|84|79.76%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|17|17|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 106 / 129 (82.17%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|67|84|79.76%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|17|17|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 107 / 129 (82.95%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|67|84|79.76%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|17|17|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AdminStmtAnalyzer.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+            }
+
+            if (!analyzeWhere(adminShowReplicaStatusStmt)) {
+                throw new SemanticException(
+                        "Where clause should looks like: status =/!= 'OK/DEAD/VERSION_ERROR/SCHEMA_ERROR/MISSING'");
+            }
+            return null;
+        }
+
+        private boolean analyzeWhere(AdminShowReplicaStatusStmt adminShowReplicaStatusStmt) {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AdminStmtAnalyzer.java:123]: miss set analyzed property to AdminShowReplicaStatusStmt

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 122 / 145 (84.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|75|93|80.65%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|17|17|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|1|1|100.00%|

**Review by HangyuanLiu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**